iothub_client/tests/iothubclient_fuzz_amqp/iothubclient_fuzz.c

```diff
@@ -0,0 +1,685 @@
+    (void)printf("deviceTwinCallback() update_state=%d\r\n", update_state);
+}
+
+int main(int argc, const char* argv[])
+{
+    IOTHUB_CLIENT_TRANSPORT_PROVIDER protocol;
+    IOTHUB_MESSAGE_HANDLE message_handle;
+    size_t messages_sent = 0;
+    const char* telemetry_msg = "test_message";
+
+    if (argc == 2)
+    {
+        test_filepath = (char*)argv[1];
+
+        FILE* fp = fopen(test_filepath, "rb");
+        if (fp == NULL)
+        {
+            (void)printf("ERROR: Cant open file %s\r\n", test_filepath);
+            return -1;
+        }
+        fuzzpacket_len = fread(fuzzpacket_buffer, sizeof(fuzzpacket_buffer[0]), sizeof(fuzzpacket_buffer), fp);
+        fclose(fp);
+
+        if (fuzzpacket_len == sizeof(fuzzpacket_buffer))
+        {
+            (void)printf("ERROR: test file is to big\r\n");
+            return -1;
+        }
+
+        if (fuzzpacket_buffer[0] != (fuzzpacket_buffer[1] ^ 0xff))
+        {
+            // packe ID got fuzzed, ok to return success
+            return 0;
+        }
+
+        fuzzpacket_id = fuzzpacket_buffer[0];
+        (void)printf("fuzzing with packet id: %d\r\n", fuzzpacket_id);
+    }
+    else if (argc > 1)
+    {
+        (void)printf("Usage:\r\n");
+        return -1;
+    }
+
+
+    // Select the Protocol to use with the connection
+#ifdef SAMPLE_MQTT
+    protocol = MQTT_Protocol;
+#endif // SAMPLE_MQTT
+#ifdef SAMPLE_MQTT_OVER_WEBSOCKETS
+    protocol = MQTT_WebSocket_Protocol;
+#endif // SAMPLE_MQTT_OVER_WEBSOCKETS
+#ifdef SAMPLE_AMQP
+    protocol = AMQP_Protocol;
+#endif // SAMPLE_AMQP
+#ifdef SAMPLE_AMQP_OVER_WEBSOCKETS
+    protocol = AMQP_Protocol_over_WebSocketsTls;
+#endif // SAMPLE_AMQP_OVER_WEBSOCKETS
+#ifdef SAMPLE_HTTP
+    protocol = HTTP_Protocol;
+#endif // SAMPLE_HTTP
+
+    // Used to initialize IoTHub SDK subsystem
+    (void)IoTHub_Init();
+
+    IOTHUB_DEVICE_CLIENT_LL_HANDLE device_ll_handle;
+
+    (void)printf("Creating IoTHub Device handle\r\n");
+    // Create the iothub handle here
+    device_ll_handle = IoTHubDeviceClient_LL_CreateFromConnectionString(connectionString, protocol);
+    if (device_ll_handle == NULL)
+    {
+        (void)printf("Failure creating IotHub device. Hint: Check your connection string.\r\n");
+    }
+    else
+    {
+        // Set any option that are neccessary.
+        // For available options please see the iothub_sdk_options.md documentation
+
+#ifndef SAMPLE_HTTP
+        // Can not set this options in HTTP
+        bool traceOn = true;
+        IoTHubDeviceClient_LL_SetOption(device_ll_handle, OPTION_LOG_TRACE, &traceOn);
+#endif
+
+#if defined SAMPLE_MQTT || defined SAMPLE_MQTT_WS
+        //Setting the auto URL Encoder (recommended for MQTT). Please use this option unless
+        //you are URL Encoding inputs yourself.
+        //ONLY valid for use with MQTT
+        //bool urlEncodeOn = true;
+        //IoTHubDeviceClient_LL_SetOption(device_ll_handle, OPTION_AUTO_URL_ENCODE_DECODE, &urlEncodeOn);
```

Comment:
if this is recommended should we have it on by default?

Reply:
This is a AMQP fuzzer, so this will never be defined. Cane from sample code. Removing